emulator.py

```diff
@@ -26,26 +39,116 @@ def __init__(self):
-            self.x_scaled = self.scaler.transform(self.x)
-            X_train, X_test, y_train, y_test = train_test_split(self.x_scaled, self.y)
+            self.scaler.fit(self.xs_train_orig)
+            self.xs_train = self.scaler.transform(self.xs_train_orig)
         else:
-            X_train, X_test, y_train, y_test = train_test_split(self.x, self.y)
-        self.model_rf = RandomForestRegressor(verbose = 0, n_jobs=-1, n_estimators=n_estimators) #n_jobs=-1 parallelises the training
-        self.model_rf.fit(X_train, y_train)
-        y_pred = self.model_rf.predict(X_test)
-        val_loss = mean_squared_error(y_test, y_pred)
-        print("Validation Loss: " + str(val_loss))
-        self.val_loss_list.append(val_loss)
+            self.xs_train = self.xs_train_orig
+
+        if normalize_y:
+            self.y_mean = np.mean(self.ys_train_orig, axis=0)
+            self.ys_train = self.ys_train_orig/self.y_mean
+        else:
+            self.ys_train = self.ys_train_orig
+        
+        infile_test = open(test_file,'rb')
+        new_dict_test = pickle.load(infile_test)
+        infile_test.close()
         
+        input_test = new_dict_test['input_data']
+        output_test = new_dict_test['output_data']
+
+        self.xs_test_orig = np.array(input_test.drop(columns=['object_id']))
+        self.ys_test_orig = np.array(output_test.drop(columns=['object_id']))    
+        if scale:
+            self.xs_test = self.scaler.transform(self.xs_test_orig)
+        else:
+            self.xs_test = self.xs_test_orig
+
+        if normalize_y:
+            self.ys_test = self.ys_test_orig/self.y_mean
+        else:
+            self.ys_test = self.ys_test_orig
+
+        self.n_params = input_train.shape[1]-1
+        self.n_values = output_train.shape[1]-1
+        self.number_test = input_test.shape[0]
+
+
+    def train(self, regressor_name, scale=True, **kwargs):
+        self.models[regressor_name]['regressors'] = np.empty(self.n_values, dtype=object)
+        train_func = None
+        if regressor_name == "RF":
+            train_func = self.train_random_forest_regressor
+        if regressor_name == "ANN":
+            train_func = self.train_ann_regressor
+
+        for j in range(self.n_values):
+            ys_train_r = self.ys_train[:,j]
+            model = train_func(x=self.xs_train, y=ys_train_r.ravel(), **kwargs)
+            self.models[regressor_name]['regressors'][j] = model
+    
+
+    # currently assumes model has a score method that takes x and y test values
+    def test(self, regressor_name):
+        assert regressor_name in self.models, f"{regressor_name} not yet trained!"
+        scores = np.empty(self.n_values)
+        for j in range(self.n_values):
+            ys_test_r = self.ys_test[:,j]
+            model = self.models[regressor_name]['regressors'][j]
+            scores[j] = model.score(self.xs_test, ys_test_r)
+        self.models[regressor_name]['scores'] = scores
```

Comment:
The GP does not have a `score` method, so this might not be the ideal way to compare these methods.

I suggest adding another function to compare the performance os the emus that take as input the true outputs and the predicted outputs only. We can do this later because I agree with you that this should be merged as soon as possible.